src/objective/multiclass_objective.hpp

```diff
@@ -24,12 +25,32 @@ class MulticlassLogloss: public ObjectiveFunction {
     label_ = metadata.label();
     weights_ = metadata.weights();
     label_int_.resize(num_data_);
-    for (int i = 0; i < num_data_; ++i){
-        label_int_[i] = static_cast<int>(label_[i]);
-        if (label_int_[i] < 0 || label_int_[i] >= num_class_) {
-            Log::Fatal("Label must be in [0, %d), but found %d in label", num_class_, label_int_[i]);
-        }
+    for (int i = 0; i < num_data_; ++i) {
+      label_int_[i] = static_cast<int>(label_[i]);
+      if (label_int_[i] < 0 || label_int_[i] >= num_class_) {
+        Log::Fatal("Label must be in [0, %d), but found %d in label", num_class_, label_int_[i]);
+      }
     }
+    label_pos_weights_ = std::vector<float>(num_class_, 1);
+    label_neg_weights_ = std::vector<float>(num_class_, 1);
+    if (is_unbalance_) {
+      std::vector<int> cnts(num_class_, 0);
+      for (int i = 0; i < num_data_; ++i) {
+        ++cnts[label_int_[i]];
+      }
+      int average_cnt = num_data_ / num_class_;
+      for (int i = 0; i < num_class_; ++i) {
+        int cnt_cur = cnts[i];
+        int cnt_other = (average_cnt - cnts[i]);
```

Comment:
should `average_cnt` be `num_data_` here? `cnt_other`, as well as `weights`, can be negative for current algorithm.